sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosRowConverter.scala

```diff
@@ -659,27 +672,18 @@ private[cosmos] class CosmosRowConverter(
         }
     }
 
-    private def getFullFidelityMetadata(objectNode: ObjectNode): Option[ObjectNode] = {
-      if (objectNode == null) {
-        None
-      } else {
-        val metadata = objectNode.get(FullFidelityChangeFeedMetadataPropertyName)
-        if (metadata != null && metadata.isObject) {
-          Some(metadata.asInstanceOf[ObjectNode])
-        } else {
-          None
-        }
+    //  Some of these APIs can be clubbed together to avoid code duplication.
+    //  However, I don't think it is worth losing code readability and code simplicity.
+    private def getAttributeNode(objectNode: ObjectNode, attributeName: String): JsonNode = {
+      objectNode.get(attributeName) match {
+        case jsonNode: JsonNode => jsonNode
+        case _ => null
       }
     }
 
-    private def parsePreviousImage(objectNode: ObjectNode): String = {
-      getFullFidelityMetadata(objectNode)
-        match {
-        case metadataNode: Some[ObjectNode] =>
-          metadataNode.get.get(PreviousImagePropertyName) match {
-            case previousImageObjectNode: ObjectNode => Option(previousImageObjectNode).map(o => o.toString).orNull
-            case _ => null
-        }
+    private def getAttributeNodeAsString(objectNode: ObjectNode, attributeName: String): String = {
```

Comment:
For this one it is ok to return String - because we want the string-mapping to pass it into Java (I know - little inconsistent -but pretty much the only acceptable place to use null in scala is to interact with Java APIs )

Reply:
Got it. will keep it as it is.